File: FCTSeek-frontend-movil/src/services/comentarioService.ts
// src/services/comentarioService.ts
import { api } from '../config/api';
import { Comentario } from '../types';

export interface ComentarioRequest {
  empresaId: number;
  texto: string;
  esPrivado: boolean;
}

export const comentarioService = {
  getByEmpresa: (empresaId: number) =>
    api.get<Comentario[]>(`/comentarios/empresa/${empresaId}`),

  crear: (data: ComentarioRequest) =>
    api.post<Comentario>('/comentarios', data),

  eliminar: (id: number) => api.delete<void>(`/comentarios/${id}`),
};

File: FCTSeek-desktop/src/main/java/com/practicas/service/ComentarioService.java
package com.practicas.service;

import com.practicas.model.Comentario;
import com.practicas.util.ApiClient;
import com.practicas.util.ApiClient.ApiException;

import java.util.List;
import java.util.Map;

public class ComentarioService {

    public static List<Comentario> getByEmpresa(long empresaId) throws ApiException {
        return ApiClient.getList("/comentarios/empresa/" + empresaId, Comentario.class);
    }

    public static Comentario crear(long empresaId, String texto, boolean esPrivado) throws ApiException {
        return ApiClient.post("/comentarios",
                Map.of("empresaId", empresaId, "texto", texto, "esPrivado", esPrivado),
                Comentario.class);
    }

    public static void eliminar(long id) throws ApiException {
        ApiClient.delete("/comentarios/" + id);
    }
}
